codegen/scope.go

```diff
@@ -65,28 +65,50 @@ done:
 	return name
 }
 
+// Unique returns a unique name for the given name. If given name not unique
```

Comment:
Might be better to specify in `HashedUnique` the fact that requesting `HashedUnique` multiple times for keys returning the same hash return the same value. And not mention anything here about it since the user of the code doesn't know what the "named and count hash" is.